File: starters/go-api/internal/server/observability_test.go
package server

import "testing"

func TestRequestLogFields_ContainsTraceAndRequestID(t *testing.T) {
	fields := RequestLogFields("GET", "/health/live", 200, "req-123", "trace-456")

	if fields["request_id"] != "req-123" {
		t.Fatalf("expected request_id, got %v", fields["request_id"])
	}
	if fields["trace_id"] != "trace-456" {
		t.Fatalf("expected trace_id, got %v", fields["trace_id"])
	}
}

File: starters/node-api/src/observability.test.ts
import { buildRequestLog } from "./server";

describe("observability log schema", () => {
  it("includes request_id and trace_id fields", () => {
    const log = buildRequestLog("GET", "/health/live", 200, "req-123", "trace-456");

    expect(log).toHaveProperty("request_id", "req-123");
    expect(log).toHaveProperty("trace_id", "trace-456");
    expect(log).toHaveProperty("http.method", "GET");
  });
});
